netket/operator/_discrete_operator.py

```diff
@@ -135,7 +135,7 @@ def n_conn(self, x, out=None) -> np.ndarray:
 
         """
         if out is None:
-            out = np.empty(x.shape[0], dtype=np.intc)
+            out = np.empty(x.shape[0], dtype=np.int32)
```

Comment:
I changed this from intc to int32 so that I can always specify int32 in JAX. This is the only usage of intc in our codebase, and I guess it's for some old C API before NetKet 3.